src/main/scala/nodes/learning/SparseLinearMapper.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nodes.learning
+
+import breeze.linalg._
+import org.apache.spark.rdd.RDD
+import workflow.Transformer
+
+/**
+ * Computes A * x + b i.e. a linear map of data using a trained model.
+ *
+ * @param x trained model
+ * @param bOpt optional intercept to add
+ */
+case class SparseLinearMapper(
```

Comment:
I'll let @etrain chime in on the scaler stuff, but I actually think it'll be good to hide most of the Sparse vs. Dense stuff from the user. i.e. if you write a Newsgroups or TIMIT pipeline you should be able to say LBFGSwithL2 as a node and not worry about whether the data comes out as sparse or dense etc. 

So some of the things like bulk apply we might need a `isDense` check and then do a different thing based on it. That way we incur some complexity in the implementation but the interface is simple.